src/txmempool.cpp

```diff
@@ -151,33 +152,47 @@ void CTxMemPool::UpdateTransactionsFromBlock(const std::vector<uint256> &vHashes
     }
 }
 
-bool CTxMemPool::CalculateMemPoolAncestors(const CTxMemPoolEntry &entry, setEntries &setAncestors, uint64_t limitAncestorCount, uint64_t limitAncestorSize, uint64_t limitDescendantCount, uint64_t limitDescendantSize, std::string &errString, bool fSearchForParents /* = true */) const
+bool CTxMemPool::CalculateMemPoolAncestors(const std::vector<CTxMemPoolEntry::CTxMemPoolEntryRef>& entries,
+                                           setEntries& setAncestors,
+                                           const uint64_t limitAncestorCount,
+                                           const uint64_t limitAncestorSize,
+                                           const uint64_t limitDescendantCount,
+                                           const uint64_t limitDescendantSize,
```

Comment:
What do you think about not making these const, and then decrementing from them and checking that we don't go below zero? That would avoid the need for the local variables and make the logic for individual txs and packages more similar.

Reply:
I think it is clearer to make these as const personally, decrement until 0 is always confusing to me (is it at 0 or below zero that the error comes in? These are currently uint64_t, so we'd also need to either convert to int64_t or detect wraparound...).

Reply:
Another problem with editing these variables is we'd need to change the error logging (current log would use the decremented count instead of actual limit), so I've left it as is.